dbt/include/duckdb/macros/materializations/incremental.sql

```diff
@@ -39,13 +41,20 @@
     {% set build_sql = create_table_as(False, intermediate_relation, compiled_code, language) %}
     {% set need_swap = true %}
   {% else %}
+    {% if not temporary %}
+      -- if not using a temporary table we will update the temp relation to use a different temp schema ("dbt_temp" by default)
+      {% set temp_relation = temp_relation.incorporate(path=adapter.get_temp_relation_path(this)) %}
+      {% do run_query(create_schema(temp_relation)) %}
+      -- then drop the temp relation after we insert the incremental data into the target relation
+      {% set need_drop_temp = True %}
```

Comment:
One last nit-- I think this variable is redundant now, right? The `{% if need_drop_temp %}` block below just needs to be `{% if not temporary %}`?

Reply:
Great catch!